components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php

```diff
@@ -0,0 +1,398 @@
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+/**
+ * @ilCtrl_isCalledBy ilAuthLogoutPageEditorGUI: ilObjAuthSettingsGUI
+ * @ilCtrl_Calls      ilAuthLogoutPageEditorGUI: ilLogoutPageGUI
+ */
+class ilAuthLogoutPageEditorGUI
+{
+    private ilCtrlInterface $ctrl;
+    private ilLanguage $lng;
+    private ilGlobalTemplateInterface $tpl;
+    private ilTabsGUI $tabs;
+    private ilToolbarGUI $toolbar;
+    private ilRbacSystem $rbacsystem;
+    private ilSetting $setting;
+    private ilErrorHandling $ilErr;
+    private ?ilPropertyFormGUI $form;
+    private \ILIAS\HTTP\Services $http;
+    private \ILIAS\Refinery\Factory $refinery;
+    private \ILIAS\UI\Factory $ui_factory;
+    private \ILIAS\UI\Renderer $ui_renderer;
+
+    private int $ref_id;
+    private ilAuthLogoutPageEditorSettings $settings;
+    private ?ilSetting $logoutSettings = null;
+    private \ILIAS\Style\Content\Object\ObjectFacade $content_style_domain;
+
+    private ?string $redirect_source = null;
+    private ?int $key = null;
+    private array $visible_languages = [];
+    private array $languages = [];
+
+    public function __construct(int $a_ref_id)
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->tpl = $DIC->ui()->mainTemplate();
+        $this->tabs = $DIC->tabs();
+        $this->toolbar = $DIC->toolbar();
+        $this->rbacsystem = $DIC->rbac()->system();
+        $this->setting = $DIC->settings();
+        $this->ilErr = $DIC['ilErr'];
+
+        $this->http = $DIC->http();
+        $this->refinery = $DIC->refinery();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
+
+        $this->lng = $DIC['lng'];
+
+        $this->lng->loadLanguageModule('auth');
+        $this->ref_id = $a_ref_id;
+
+        $this->settings = ilAuthLogoutPageEditorSettings::getInstance();
+        $this->content_style_domain = $DIC->contentStyle()
+                                          ->domain()
+                                          ->styleForRefId($a_ref_id);
+
+        $query_wrapper = $DIC->http()->wrapper()->query();
+        $post_wrapper = $DIC->http()->wrapper()->post();
+        $is_post_request = $DIC->http()->request()->getMethod() === 'POST';
+        $refinery = $DIC->refinery();
+
+        if ($query_wrapper->has('redirectSource')) {
+            $this->redirect_source = $query_wrapper->retrieve('redirectSource', $refinery->kindlyTo()->string());
+        }
+
+        if ($post_wrapper->has('key')) {
+            $this->key = $post_wrapper->retrieve('key', $refinery->kindlyTo()->int());
+        } elseif ($query_wrapper->has('key')) {
+            $this->key = $query_wrapper->retrieve('key', $refinery->kindlyTo()->int());
+        }
+
+        if ($is_post_request) {
```

Comment:
Do we really need the condition and the variable `is_post_request`?

Reply:
done